In [48]:
import pandas as pd
import numpy as np



Dataset is taken from https://www.kaggle.com/avikasliwal/used-cars-price-prediction?select=train-data.csv


In [49]:
file_url = "/content/train-data.csv"

In [50]:
df = pd.read_csv(file_url)
df.head()

Unnamed: 0                              Name  ...  New_Price  Price
0           0            Maruti Wagon R LXI CNG  ...        NaN   1.75
1           1  Hyundai Creta 1.6 CRDi SX Option  ...        NaN  12.50
2           2                      Honda Jazz V  ...  8.61 Lakh   4.50
3           3                 Maruti Ertiga VDI  ...        NaN   6.00
4           4   Audi A4 New 2.0 TDI Multitronic  ...        NaN  17.74

[5 rows x 14 columns]

We are going to extract the Kilometers_Driven column and find the Quantitle value which divide it into 2 group

In [51]:
kilo_driven = df["Kilometers_Driven"]
 
quantile_values = np.quantile(kilo_driven,[0.5])

quantile_values

array([53000.])

Quantile value (the cut off value) is 53000

In [52]:
#divide by 50% quantile value

group1 = kilo_driven[kilo_driven<quantile_values[0]]
group2 = kilo_driven[kilo_driven>quantile_values[0]]

In [53]:
total_row = kilo_driven.shape[0]

group1_row = group1.shape[0]

group2_row = group2.shape[0]

print(f"Total row count:{total_row}")
print(f"Group 1 row count:{group1_row}")
print(f"Group 2 row count:{group2_row}")
print(f"Group 1+2 row count:{group1_row+group2_row}")

Total row count:6019
Group 1 row count:2987
Group 2 row count:3007
Group 1+2 row count:5994


Notice that we are missing some row since 2987 + 3007 = 5994 and the total row count is 6019

In [54]:
group1_percentage = (group1_row/total_row)*100

group2_percentage = (group2_row/total_row)*100

print(f"Group 1 percentage:{group1_percentage}")
print(f"Group 2 percentage:{group2_percentage}")
print(f"Group 1+2 percentage:{group1_percentage+group2_percentage}")

Group 1 percentage:49.62618375145373
Group 2 percentage:49.95846486127264
Group 1+2 percentage:99.58464861272637


Like group row count , we are missing some percentage since 49.62 + 49.95 = 99.58% and it is not 100%

Let find the missing 25 row count since 6019-5994 = 25

In [55]:
cut_off_row = kilo_driven[kilo_driven==quantile_values[0]].shape[0] 

cut_off_row

25

In [56]:
print(f"Total row count:{group1_row+group2_row+cut_off_row}")

Total row count:6019


As seen above the missing 25 row is the row which have the exact quantile value (53000)